eclipse/plugins/org.switchyard.tools.ui.editor/src/org/switchyard/tools/ui/editor/components/camel/xml/NewCamelXMLRouteCompenentWizard.java

```diff
@@ -92,13 +108,70 @@ String getSpringStubText(String serviceName) {
         sb.append("</beans>\n");
         return sb.toString();
     }
-    
+
+    private InputStream getOldRouteStubText() {
+        ByteArrayOutputStream baos = null;
+        try {
+            DocumentRoot doc = SpringFactory.eINSTANCE.createDocumentRoot();
+            RoutesDefinition routes = SpringFactory.eINSTANCE.createRoutesDefinition();
+            RouteDefinition route = SpringFactory.eINSTANCE.createRouteDefinition();
+            route.setId("_route1");
+            FromDefinition from = SpringFactory.eINSTANCE.createFromDefinition();
+            from.setId("_from1");
+            LogDefinition log = SpringFactory.eINSTANCE.createLogDefinition();
+            log.setId("_log1");
+            String serviceName = getService().getName();
+
+            from.setUri("switchyard://" + serviceName); //$NON-NLS-1$
+            log.setMessage("" + serviceName + " - message received: ${body}"); //$NON-NLS-1$ //$NON-NLS-2$
+
+            doc.setRoutes(routes);
+            doc.setRoute(route);
+            route.getFrom().add(from);
+            route.getLog().add(log);
+            routes.getRoute().add(route);
+
+            doc.getXMLNSPrefixMap().put("", SpringPackage.eNS_URI); //$NON-NLS-1$
+
+            Resource routeResource = new SpringResourceFactoryImpl().createResource(URI.createGenericURI("temp", //$NON-NLS-1$
+                    "temp", null)); //$NON-NLS-1$
+            routeResource.getContents().add(doc);
+
+            baos = new ByteArrayOutputStream();
+            routeResource.save(baos, null);
+            return new ByteArrayInputStream(baos.toByteArray());
+        } catch (IOException e2) {
+            e2.printStackTrace();
+            Activator.logError(e2);
+        } finally {
+            if (baos != null) {
+                try {
+                    baos.close();
+                } catch (IOException e) {
+                    e.fillInStackTrace();
+                }
+            }    
+        }
+        return null;
+    }
     
     @Override
     protected InputStream getInitialContents() {
+        boolean useNewCamelRoute = true;
+        IProject project = getJavaProject().getProject();
+        String version = SwitchYardModelUtils.getSwitchYardProjectRuntimeVersion(project);
+        if (version.startsWith("2.0") || version.startsWith("1")) {
+            useNewCamelRoute = false;
+        }
+        // otherwise we're using the new one
+        
         String serviceName = getService().getName();
         if (serviceName != null) {
-            return new ByteArrayInputStream(getSpringStubText(serviceName).getBytes());
+            if (useNewCamelRoute) {
+                return new ByteArrayInputStream(getNewRouteStubText(serviceName).getBytes());
```

Comment:
better to precise encoding in toBytes
We often use the UTF-8 by default instead of the system one at least
java.nio.charset.StandardCharsets.UTF_8 is useful to get the UTF-8 charset